ui/src/pages/version/HistoryDetails.js

```diff
@@ -0,0 +1,182 @@
+  const orderedDeployments = deployments.sort((a, b) => b.id - a.id);
+
+  const deployedRevision = orderedDeployments.find(
+    (deployment) =>
+      deployment.status === "running" || deployment.status === "serving"
+  ) || { id: null };
+
+  const canBeExpanded = (deployment) => {
+    return deployment.error !== "";
+  };
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState({});
+
+  const toggleDetails = (deployment) => {
+    const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+    if (itemIdToExpandedRowMapValues[deployment.id]) {
+      delete itemIdToExpandedRowMapValues[deployment.id];
+    } else {
+      itemIdToExpandedRowMapValues[deployment.id] = (
+        <>
+          <EuiText className="expandedRow-title" size="xs">
+            Error message
+          </EuiText>
+          <EuiCodeBlock isCopyable>{deployment.error}</EuiCodeBlock>
+        </>
+      );
+    }
+    setItemIdToExpandedRowMap(itemIdToExpandedRowMapValues);
+  };
+
+  const cellProps = (item, column) => {
+    if (column.field !== "actions" && canBeExpanded(item)) {
+      return {
+        style: { cursor: "pointer" },
+        onClick: () => toggleDetails(item),
+      };
+    }
+    return undefined;
+  };
+
+  const columns = [
+    {
+      field: "updated_at",
+      name: "Deployment Time",
+      render: (date, deployment) => (
+        <>
+          <DateFromNow date={date} size={defaultTextSize} />
+          &nbsp;&nbsp;
+          {deployment.id === deployedRevision.id && (
+            <EuiBadge color="default">Current</EuiBadge>
+          )}
+          {/* {JSON.stringify(deployment.id)} */}
+        </>
+      ),
+    },
+    {
+      field: "status",
+      name: "Deployment Status",
+      render: (status, deployment) => (
+        <DeploymentStatus
+          status={status}
+          deployment={deployment}
+          deployedRevision={deployedRevision}
+          endpointStatus={endpoint.status}
+        />
+      ),
+    },
+    {
+      align: "right",
+      width: "40px",
+      isExpander: true,
+      name: (
+        <EuiScreenReaderOnly>
+          <span>Expand rows</span>
+        </EuiScreenReaderOnly>
+      ),
+      render: (deployment) => {
+        const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+        return (
+          canBeExpanded(deployment) && (
+            <EuiButtonIcon
+              onClick={() => toggleDetails(deployment)}
+              aria-label={
+                itemIdToExpandedRowMapValues[deployment.id]
+                  ? "Collapse"
+                  : "Expand"
+              }
+              iconType={
+                itemIdToExpandedRowMapValues[deployment.id]
+                  ? "arrowUp"
+                  : "arrowDown"
+              }
+            />
+          )
+        );
+      },
+    },
+  ];
+
+  return (
+    <ConfigSection title="Deployment History">
+      <ConfigSectionPanel>
+        <EuiInMemoryTable
```

Comment:
Shall we enable pagination on the table? It can help if there have been too many redeployments.

Reply:
Agree. Already added `pagination={true}` prop below.